### PR TITLE
active directory client is more generic

### PR DESCRIPTION
**Pull request by dtourek**
search function is now working and tested
Eslint fixes implementation

### CODE REVIEW

File: src/activeDirectory/activeDirectoryServer.ts [hunk 1]
```diff
@@ -1,101 +1,95 @@
...
-        if (!req.dn.equals(args.bindDN) || req.credentials !== args.bindPassword) {
-            return next(new ldap.InvalidCredentialsError());
-        }
-
-        res.end();
-        return next();
-    });
+  suffix
+    .split(',')
+    .map((a) => a.trim())
```
Comment (kolebjak) [inline src/activeDirectory/activeDirectoryServer.ts:12]: unnecessary double map. could be handled in one iteration.

File: src/activeDirectory/activeDirectoryServer.ts [hunk 2]
```diff
@@ -1,101 +1,95 @@
...
-        const dn = req.dn.toString();
-        logger('info', 'SUFFIX bind for', { dn });
+export const ActiveDirectoryServer = (adArgs: { bindDN: string; bindPassword: string; suffix: string; users: IUser[]; logger?: (...args: any[]) => void }) => {
+  const server = ldap.createServer();
 
-        const user = args.users.find((u) => {
-            const x = `cn=${u.givenName} ${u.sn}, ou=Users, ${lowercaseDC(args.suffix)}`;
-            return dn === x && u.password === req.credentials;
-        });
+  const logger = adArgs.logger ? adArgs.logger : () => null;
```
Comment (kolebjak) [inline src/activeDirectory/activeDirectoryServer.ts:19]:
```suggestion
  const logger = adArgs.logger ? adArgs.logger : console.log;
```

File: src/activeDirectory/activeDirectoryServer.ts (hunk 1 above)
Comment (dtourek) [reply src/activeDirectory/activeDirectoryServer.ts]: fix in https://github.com/flsy/ldap-stub/pull/5/commits/ef3b2580faee13216d55bac8b7c29755f4ebc689

File: src/activeDirectory/activeDirectoryServer.ts (hunk 2 above)
Comment (dtourek) [reply src/activeDirectory/activeDirectoryServer.ts]: fix in https://github.com/flsy/ldap-stub/pull/5/commits/ef3b2580faee13216d55bac8b7c29755f4ebc689

File: src/activeDirectory/activeDirectoryServer.ts [hunk 3]
```diff
@@ -1,101 +1,97 @@
...
-        }
-
-        res.end();
-        return next();
-    });
+  suffix
+    .split(',')
+    .map((value) => {
+      const dc = value.trim().toLowerCase();
+      return dc.startsWith('dc=') ? dc : value;
```
Comment (smidhonza) [inline src/activeDirectory/activeDirectoryServer.ts:14]: ted se ta funkce chova ale trochu jinak, ne?

predtim:
`lovercaseDC('DC=Moje') vratilo 'dc=Moje';`

ted:
`lovercaseDC('DC=Moje') vrati 'dc=moje';`

je to tak dobre ?

**Review by smidhonza — CHANGES_REQUESTED** [review]
Ahoj, prosimte rozdel ten pull request na dva, jeden kde mas ty zmeny co chces udelat a na druhej kde celej kod menis kvuli lintu a prettieru. Tomuhle se vazne neda rozumne udelat review kdyz kombinujes milion zmen do jednoho PR. dik

Vubec neni jasny o jaky zmeny tu vubec jde.

File: src/activeDirectory/activeDirectoryServer.ts (hunk 3 above)
Comment (dtourek) [reply src/activeDirectory/activeDirectoryServer.ts:14]: jo chapu, on ten textovy retezec jeste pokracuje

**Pull request closed without merging**